pkg/client/dtclient/config_client.go

```diff
@@ -485,7 +411,48 @@ func isUserSessionPropertiesMobile(a api.API) bool {
 	return a.ID == api.UserActionAndSessionPropertiesMobile
 }
 
-func (d *DynatraceClient) getExistingValuesFromEndpoint(ctx context.Context, theApi api.API, urlString string) (values []Value, err error) {
+func (d *DynatraceClient) getExistingObjectId(ctx context.Context, objectName string, theApi api.API, payload []byte) (string, error) {
+	// if there is a custom equal function registered, use that instead of just the Object name
+	// in order to search for existing values
+	if theApi.CheckEqualFunc != nil {
+		values, err := d.fetchExistingValues(ctx, theApi, theApi.CreateURL(d.environmentURLClassic))
+		if err != nil {
+			return "", err
+		}
+		return d.findUnique(values, payload, theApi.CheckEqualFunc)
+	}
+
+	// Single configuration APIs don't have an id which allows skipping this step
+	if !theApi.SingleConfiguration {
+		values, err := d.fetchExistingValues(ctx, theApi, theApi.CreateURL(d.environmentURLClassic))
+		if err != nil {
+			return "", err
+		}
+		return d.findUniqueByName(ctx, theApi, values, objectName)
+
+	}
+	// The network-zone API doesn't have a POST endpoint, hence, we need to treat it as an update operation
+	// per default
+	if theApi.ID == api.UserActionAndSessionPropertiesMobile {
```

Comment:
as these are now all returning directly, can you please structure this method so that they are early exit conditions? 

only if none of these ifs match - and the API is not a `SingleConfiguration` one (make early exit too), do we need to make any API calls to fetch existing values.

Suggestion: 

```go
if theApi.ID == api.UserActionAndSessionPropertiesMobile { return objectName }
if theApi.ID == api.NetworkZone { return objectName }
if theApi.ID == api.CalculatedMetricsLog { return objectName }
if theApi.SingleConfiguration { return "" }

values, err := d.fetchExistingValues(ctx, theApi, theApi.CreateURL(d.environmentURLClassic))
if err != nil {
	return "", err
}

if theApi.CheckEqualFunc != nil { return d.FindUnique } 

return d.FindUniqueByName
```